sky/backends/backend_utils.py

```diff
@@ -1607,8 +1608,56 @@ def _query_status_azure(
 }
 
 
+def get_cloud_user_identity_no_error(cloud: clouds.Cloud) -> Optional[str]:
+    try:
+        return cloud.get_current_user_identity()
+    except exceptions.CloudUserIdentityError as e:
+        logger.warning(
+            f'Failed to get the current user identity for {cloud}: {e}')
+
+
+def check_owner_identity(cluster_name: str):
+    """Check if the current user is the same as the user who created the cluster.
+
+    This should be called before any operation that will modify the cluster, or query
+    from the cloud provider.
+
+    Raises:
+        exceptions.ClusterOwnerIdentityMismatchError: if the current user is not the
+            same as the user who created the cluster.
+    """
+    if env_options.Options.SKIP_CLOUD_IDENTITY_CHECK.get():
+        return
+    record = global_user_state.get_cluster_from_name(cluster_name)
+    if record is None:
+        return
+    handle = record['handle']
+    if not isinstance(handle, backends.CloudVmRayBackend.ResourceHandle):
+        return
+
+    cloud = handle.launched_resources.cloud
+    current_user_identity = get_cloud_user_identity_no_error(cloud)
+    owner_identity = record['owner']
+    if current_user_identity is None:
+        # Skip the check if the cloud does not support user identity,
+        # or we fail to get the user identity.
```

Comment:
When we fail to get identity, why is it ok to let this operation go through? See another comment about potential leakage.

In the case of "if the cloud does not support user identity" it may be ok; likely should check any recorded owner = None.

Reply:
I was intended to make this check minimum, so that it just serves as a minimum safeguard, but not panic for failing to get the owner identity.
However, it makes sense that this can cause leakage. Let me revert it to using the `get_current_user_identity` that will raise error (also for the one in `backend.provision()`).

Reply:
Sounds good; should we remove L1663?

Reply:
Removed. Thanks!